controllers/bpfman-agent/common.go

```diff
@@ -71,6 +72,9 @@ type ReconcilerCommon struct {
 	Logger       logr.Logger
 	NodeName     string
 	progId       *uint32
+	finalizer    string
+	recType      string
+	appOwner     metav1.Object // Set if the owner is an application
```

Comment:
wait why do we need to persist finalizer and recType? The're both always just constants right? (even for the new BpfApplication object)

appOwner I could forsee being a bit different since it's either the

Reply:
Because when we're creating BpfProgram objects for a BpfApplication, they need to have the BpfApplication finalizer and recType.  To make this happen, I set these when the Reconciler is initialized so the getter can return the right one.

Reply:
Ok I see, that works